src/Redux/actionCreators/UserActions.js

```diff
@@ -129,3 +133,97 @@ export const checkWalletStatus = (dispatch, getState) => {
       }
     });
 };
+
+const userDeleted = ({ history, route }) => dispatch => {
+  history.push(route);
+  dispatch({
+    type: SET_USER_DETAILS,
+    payload: {
+      login: {
+        isLoggedIn: false,
+      },
+      isEmailVerified: false,
+      isWalletAssigned: false,
+      email: "",
+    },
+  });
+};
+const deleteUser = user =>
+  new Promise((resolve, reject) => {
+    user.deleteUser(error => {
+      if (error) {
+        reject(error);
+      }
+      resolve();
+    });
+  });
+
+const fetchCurrentUser = () => {
+  return Auth.currentAuthenticatedUser({ bypassCache: true });
+};
+
+export const deleteUserAccount = ({ history, route }) => dispatch => {
+  dispatch(() =>
+    fetchCurrentUser().then(user => {
+      dispatch(() =>
+        deleteUser(user).then(() => {
+          dispatch(userDeleted({ history, route }));
+        })
+      );
+    })
+  );
+};
+
+const forgotPasswordInit = dispatch => {
+  dispatch(loaderActions.startAppLoader);
+  dispatch(errorActions.resetForgotPasswordError);
+};
+
+const forgotPasswordSuccessfull = ({ username, history, route }) => dispatch => {
+  dispatch(updateUsername(username));
+  dispatch(loaderActions.stopAppLoader);
+  history.push(route);
+};
+
+const forgotPasswordFailure = error => dispatch => {
+  dispatch(loaderActions.stopAppLoader);
+  dispatch(errorActions.updateForgotPasswordError(error));
+};
+
+export const forgotPassword = ({ username, history, route }) => dispatch => {
+  dispatch(forgotPasswordInit);
+  Auth.forgotPassword(username)
+    .then(() => {
+      dispatch(forgotPasswordSuccessfull({ history, route }));
+    })
+    .catch(err => {
+      dispatch(forgotPasswordFailure(err.message));
+    });
+};
+
+const forgotPasswordSubmitInit = dispatch => {
+  dispatch(loaderActions.startAppLoader);
+  dispatch(errorActions.resetForgotPasswordSubmitError);
+};
+
+const forgotPasswordSubmitSuccessfull = ({ username, history, route }) => dispatch => {
+  dispatch(updateUsername(username));
+  dispatch(loaderActions.stopAppLoader);
+  history.push(route);
```

Comment:
👍 
This looks like a correct sequence of operations. 
1. Update state
2. Stop loader
3. Navigate.

Reply:
Thanks :)